### ISSUE TITLE
VCOCO setting

### ISSUE THREAD

**Issue by ChelsieLei** [ltttpku/ADA-CM#8]
Hi, thank you for your nice work. 

Could you please provide the process to evaluate the VCOCO? 

Your script can generate the cached results for V-COCO, but then when I run the eval_coco.py, by using your pre-trained weights, the number is 
Average Role [scenario_1] AP = 56.20 , Average Role [scenario_2] AP = 61.39

I think my steps to obtain the results are problematic, but I don't know where the problem exists.
The following are my steps to obtain the results. 
1.  python main_tip_finetune.py --world-size 1 --dataset vcoco --data-root vcoco/ --partitions trainval test --pretrained checkpoints/detr-r50-vcoco.pth --output-dir checkpoints/vcoco  --use_insadapter  --num_classes 24 --use_multi_hot --file1 vcoco_pkl_files/vcoco_union_embeddings_cachemodel_crop_padding_zeros_vit336.p    --clip_dir_vit   checkpoints/pretrained_CLIP/ViT-L-14-336px.pt  --cache --resume   chekcpoints/vcoco/ada_cm_vcoco_vit336px.pt

2. python eval_vcoco.py /checkpoints/vcoco/cache.pkl

Thank you very much if you can help me with this issue!

**Comment by ltttpku** [comment]
Thanks for your interest in our work!

Please refer to _**step 4**_ of these [instructions](https://github.com/fredzzhang/upt/discussions/14).
The average precision calculation incorporates a class (`point-instr`) that does not correspond to valid human-object interactions, hence it's excluded from the final performance as per convention. You may need to multiply the computed value by 25 and then divide by 24 to get the true performance.

**Comment by ChelsieLei** [comment]
Thanks very much! Got it.

**Comment by ChelsieLei** [comment]
Thanks very much! Got it.

**Comment by ChelsieLei** [comment]
Sorry to bother you again. I wonder why in the vcoco_text_label.py,  vcoco_hoi_text_label (https://github.com/ltttpku/ADA-CM/blob/main/vcoco_text_label.py#L327) doesn't contain actions such as running and pointing, but they are included in the vcoco evaluation. 
 I saw that you commented out a previous version of vcoco_hoi_text_label, which contains all actions. (https://github.com/ltttpku/ADA-CM/blob/main/vcoco_text_label.py#L44)  However, the action labels are different from the previous one. May I know which one should I use?

**Comment by ChelsieLei** [comment]
Oh, I got it. In scenarios 1 and 2, there is no running or smiling, while the pointing is dismissed by *25/24 process indicated by your last reply. 
However, I have another question, what is the meaning or difference between "agent", "obj" and "instr". I guess "agent" is the subject and "obj" is the object, but how about the "instr"?
Thank you very much!

**Comment by ltttpku** [comment]
Hey, sorry for the late reply. The `instr` refers to the instrument involved in the interaction. You can refer to the [V-COCO paper](https://arxiv.org/pdf/1505.04474v1) (blue box in Fig. 1) for more details.

**Comment by ChelsieLei** [comment]
No problem. Thank you for your reply!
